org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java

```diff
@@ -314,4 +318,12 @@ public JavaVersion getJavaVersion() {
 		JavaVersion version = JavaVersion.fromBree(selected);
 		return version;
 	}
+	
+	@Override
+	public void setVisible(boolean visible) {
+		super.setVisible(visible);
+		if (visible) {
+			validatePage();
```

Comment:
Why do this is setVisible? Looks like an unexpected side effect here. Is there another place better suited to call it?
I guess the intention is here to validate when the page becomes active.

Reply:
I don’t know the correct place to do that on there is nobody available to tell which one is better

Reply:
Adding `validatePage()` to `createControl` seems to do the same job. It is just the initial validation that you want to trigger, all further validation is already handled by editing the page.

Reply:
thisis about page flipping